assets/js/modules/analytics-4/datastore/settings.js

```diff
@@ -222,3 +227,31 @@ export function validateCanSubmitChanges( select ) {
 		);
 	}
 }
+
+/**
+ * Gets the value of propertyCreateTime converted to UNIX timestamp.
+ *
+ * @since n.e.x.t
+ *
+ * @return {number|undefined} Value converted to the UNIX timestamp in ms. `undefined` if not loaded.
+ */
+export const getPropertyCreateTime = createRegistrySelector(
```

Comment:
I think you may have misunderstood my suggestion in https://github.com/google/site-kit-wp/pull/7894#discussion_r1408368540 where we'd decorate the action to convert the given timestamp if it was the ISO string format, to an integer. Internally it would then call the original action with the transformed value.

Overriding the selector would work too but since we're already transforming it in the backend and storing as an integer in module settings, and no longer converting it in the action, it's possible that the state for this setting would be in different formats, even though it would always be selected as the timestamp.

Here's what I had in mind by decorating the action – this way the selector would keep the original. WDYT?

```js
const originalSetPropertyCreateTime = baseStore.actions.setPropertyCreateTime;
baseStore.actions.setPropertyCreateTime = ( value ) => {
	return originalSetPropertyCreateTime(
		convertDateStringToUNIXTimestamp( value )
	);
}
```

Reply:
Aha, I see what you meant. Yes I guess it makes more sense to ensure data is saved always in same format, then converting it on return.

I have updated the resolver to use this setter for saving the value, as otherwise yielding to reducer will bypass it, and since we should stay away from overriding the reducer, setter is only remaining option.

Reply:
> I have updated the resolver to use this setter for saving the value, as otherwise yielding to reducer will bypass it, and since we should stay away from overriding the reducer, action is the only remaining option.

Correct – we generally don't yield raw actions using object literals like which are defined elsewhere either (we only do this in the same partial where the resolver is defined). Instead, (you can see some examples of this throughout as well) we import the `actions` from the store (if defined in another partial) and `yield` the result of the function. This will always dispatch the action on the current store so it only works in the context of the same top-level store (e.g. `modules/analytics-4`). This is preferable over the `registry.dispatch` form when possible because yielding raw actions will be "awaited" whereas the latter produces a `Promise` and will not.